containers/test/test_container_rpc.sh

```diff
@@ -1,34 +1,60 @@
-trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+#!/usr/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+success=false
+
+function clean_up {
+    # Perform program exit housekeeping
+    # echo Background jobs: $(jobs -l)
+    # echo
+    # echo Killing jobs
+    echo Exiting...
+    if [ "$success" = false ] ; then
+      echo 'Error in RPC test'
+    fi
+    kill $(jobs -p) &> /dev/null
+    echo
+    sleep 2
+    # echo Remaining background jobs: $(jobs -l)
+    exit
+}
+
+
+trap clean_up SIGHUP SIGINT SIGTERM EXIT
```

Comment:
Can you explain how clean_up is invoked via one of these traps when `rpctest` fails? It seems like we're trapping EXIT in this case, right?